### PR TITLE
Relaxing location data type restriction.

### PR DESCRIPTION
**Pull request by dskkato**
Currently, `folium.Map(location)` requires location data to be an instance of _list_ or _tuple_. I felt it to be inconvenient since neither numpy's array nor pandas' DataFrame cannot be used  as a _location_ argument directly. In this PR, I'm trying to relax the restriction so that any iterable object with `len(location)==2` can be accepted.

### CODE REVIEW

**Comment by ocefpaf** [comment]
You can ignore Appveyor b/c I just merged the PR that implements it. Or rebase against latest master to get the tests going on Windows. Also, the failing notebook test seems to be a network issue. However, your  change is not Python 2.7 compatible and we need a workaround there.

**Comment by dskkato** [comment]
I saw Jinja2 has similar workroud. I will try and test on Python2.7.
```
try:
    from collections import abc
except ImportError:
    import collections as abc
```
https://github.com/pallets/jinja/blob/7e417c5c6ec90709f15e96a977ab59a5397a08fb/jinja2/_compat.py#L102-L105

**Comment by ocefpaf** [comment]
> I saw Jinja2 has similar workroud. I will try and test on Python2.7.
> 
> ```
> try:
>     from collections import abc
> except ImportError:
>     import collections as abc
> ```
> [pallets/jinja:jinja2/_compat.py@`7e417c5`#L102-L105](https://github.com/pallets/jinja/blob/7e417c5c6ec90709f15e96a977ab59a5397a08fb/jinja2/_compat.py#L102-L105)


That is probably OK until we drop support for legacy python.

**Comment by dskkato** [comment]
I would like to use [Collection](https://docs.python.org/3.7/library/collections.abc.html#collections.abc.Collection) for validation, but it's only supported 3.6 or later.  So I added one helper function for the compatibility.

File: tests/test_folium.py [hunk 1]
```diff
@@ -197,7 +209,8 @@ def test_map_build(self):
 
         # Standard map.
         self.setup()
-        rendered = [line.strip() for line in self.m._parent.render().splitlines() if line.strip()]
+        rendered = [line.strip()
```
Comment (ocefpaf) [inline tests/test_folium.py]: Please avoid style changes in the same PR. It makes the reviewing process a little bit harder.
(Style changes are sometimes welcomed though, but preferably in a different PR.)

File: tests/test_folium.py [hunk 2]
```diff
@@ -62,6 +63,17 @@ def test_get_templates():
     assert isinstance(env, jinja2.environment.Environment)
 
 
+def test_location_args():
+    """Test some data types for a location arg."""
+    location = np.array([45.5236, -122.6750])
+    m = folium.Map(location)
+    assert m.location == [45.5236, -122.6750]
+
+    df = pd.DataFrame({"location": [45.5236, -122.6750]})
```
Comment (ocefpaf) [inline tests/test_folium.py]: :+1:

File: tests/test_folium.py (hunk 1 above)
Comment (dskkato) [reply tests/test_folium.py]: I reverted them. My formatter accidentally modified it...

**Comment by ocefpaf** [comment]
This is good to go IMO but I'll let it here for a few days so @Conengmo can take a second look. Planning on merging it next week.

Thanks @dskkato!

**Comment by ocefpaf** [comment]
Ooops. Wrong button! Sorry! Well... Now it is merge :smile:

**Comment by Conengmo** [comment]
Looks good! Thanks for your contribution @dskkato and Filipe for the review.

**Comment by ocefpaf** [comment]
And sorry again for the premature merge! I had way too many tabs open for a Saturday morning.

**Comment by Conengmo** [comment]
No problem

**Pull request merged**